src/main/java/com/microsoft/azure/functions/annotation/SignalRConnectionInfoInput.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+ package com.microsoft.azure.functions.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * <p>Place this on a parameter to obtain a SignalRConnectionInfo object.
+ * The parameter type can be one of the following:</p>
+ *
+ * <ul>
+ *     <li>SignalRConnectionInfo type</li>
+ * </ul>
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.PARAMETER)
+public @interface SignalRConnectionInfoInput {
+    /**
+     * The variable name used in function.json.
+     * @return The variable name used in function.json.
+     */
+    String name();
+
+    /**
+     * <p>Defines how Functions runtime should treat the parameter value. Possible values are:</p>
+     * <ul>
+     *     <li>"": get the value as a string, and try to deserialize to actual parameter type like POJO</li>
+     *     <li>string: always get the value as a string</li>
+     *     <li>binary: get the value as a binary data, and try to deserialize to actual parameter type byte[]</li>
+     * </ul>
+     * @return The dataType which will be used by the Functions runtime.
+     */
+    String dataType() default "";
+
+    /**
+     * Defines the app setting name that contains the Azure SignalR Service connection string.
+     * @return The app setting name of the connection string.
+     */
+    String connectionStringSetting() default "";
```

Comment:
Unfortunately as of we cannot support a type like this for connection string. This would mean a hard dependency on types defined here on maven plugin

Reply:
Can you explain a bit more? Why would this add a dependency on the Maven plugin?

Reply:
@jdneo / @Flanker32  - Have more context on this.
From my understanding to generate function.json, maven needs a reference for these types. @jdneo / @Flanker32  have any other ideas on how to support this type?

Reply:
Looks like if the new annotation is extended based on the `@CustomBinding`, then there is no need to add a new dependency on the plugin side.